tests/scripts/migrate-tags.py

```diff
@@ -0,0 +1,327 @@
+                    # so need to fall into the code below which finds
+                    # the test name from the line that we just read
+
+                    if test_fullpath is None:
+                        fatal('format error - RelativePath entry is missing');
+
+                    # we must record the current test information:
+                    # create the four-tuple entry
+                    entry = (test_name,     test_number, 
+                             test_fullpath, test_properties)
+
+                    testdata.append(entry)
+
+                    # reset the test state variables to empty for the next test
+                    test_name = None
+                    test_number = None
+                    test_fullpath = None
+                    test_properties = []
+
+                    # change the state to expecting_testname
+                    expecting_metadata = False
+                    expecting_testname = True
+    
+            if expecting_testname:
+                # We are expecting the next testname entry
+                # We will skip lines until we find a test name line
+                # which comes in as [ testname_number ]
+                split_line = line.split('[')
+    
+                # if we don't have a '[' the the len(split_line) will be 1
+                # we will skip (ignore) this line
+                if len(split_line) == 1:
+                    continue
+
+                if test_name is not None:
+                    fatal('logic error - test_name is already set');
+
+                if test_number is not None:
+                    fatal('logic error - test_number is already set');
+
+                # we now expect to match '[testname_number]'
+                # only when len(split_line) is 2 did we match exactly one '['
+                if len(split_line) != 2:
+                    fatal('syntax error - multiple [');
+    
+                split_line = split_line[1].split(']')
+    
+                # only when len(split_line) is 2 did we match exactly one ']'
+                if len(split_line) != 2:
+                    fatal('syntax error - missing or multiple ]:' + line);
+    
+                # get the string enclosed by [ ... ]
+                name_and_number = split_line[0]
+    
+                split_line = name_and_number.split('_')
+    
+                # Note that the testname portion may also contain '_' so we
+                # have to get the testnumber from the end using [-1]
+                if len(split_line) == 1:
+                    fatal('syntax error - missing _' + line);
+    
+                # elements in split_line are numbered [ 0, 1, ... -2, -1 ]
+                test_number_str = split_line[-1]
+                if len(split_line) == 2: 
+                    test_name = split_line[0] 
+                else:
+                    # test_name = '_'.join(split_line[0:len(split_line)-1]) 
+                    test_name = '_'.join(split_line[0:-1]) 
```

Comment:
Thank you for making this change, it is more readable because it follows convention.